h/views/groups.py

```diff
@@ -112,23 +114,12 @@ def _update_group(self, appstruct):
 @view_config(route_name='group_read',
              request_method='GET',
              renderer='h:templates/groups/share.html.jinja2',
-             effective_principals=security.Authenticated)
+             effective_principals=security.Authenticated,
+             custom_predicates=[custom_predicates.has_read_permission,
+                                not_(custom_predicates.feature('search_page'))])
 def read(group, request):
-    """Group view for logged-in users."""
-    _check_slug(group, request)
-
-    # If the current user is not a member of the group, they will not have the
-    # 'read' permission on the group. In this case, we show them the join
-    # page.
-    if not request.has_permission('read'):
-        request.override_renderer = 'h:templates/groups/join.html.jinja2'
-        return {'group': group}
-
-    # Redirect to new group overview page if search page is enabled
-    if request.feature('search_page'):
-        url = request.route_path('activity.group_search', pubid=group.pubid)
-        return HTTPSeeOther(url)
```

Comment:
We no longer redirect to the group `/search` page - instead the search page is shown at the same URL (by changing the view config rules to have Pyramid select a different view function than this one)